xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -143,6 +144,41 @@ impl<
 	}
 }
 
+/// Generates a unique `AccountId` for a given parachain + account combination.
+///
+/// Used for giving parachains control over some accounts on the interpreting chain.
+///
+/// Note: Not reversible.
+pub struct DerivedParachainAccountId32<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<
+	Network: Get<NetworkId>,
+	AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone,
+> Convert<MultiLocation, AccountId> for DerivedParachainAccountId32<Network, AccountId> {
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		use Junction::*;
+		use MultiLocation::*;
+		let id = match location {
+			X2(Parachain(para_id), AccountId32 { id, network })
+				if network == NetworkId::Any || &network == &Network::get() =>
+			{
+				(para_id, id).using_encoded(blake2_256)
```

Comment:
Should have a disabusing prefix.

Reply:
Like generate a 28byte array and prepend a u32 with the para id?